### ISSUE TITLE
Add function generate list of valid primitives, given an EntitySet

### ISSUE THREAD

**Issue by gsheni** [alteryx/featuretools#1453]
- As a user of Featuretools, I wish there was a utility function in Featuretools, which given an EntitySet, which provide me with a list of valid/applicable primitives. 
- For an EntitySet with a single DataFrame, this should be possible by looking at the typing information

```python
def get_valid_primitives(entityset, target_entity, max_depth=2, selected_primitives=None):
    """
        max_depth (int, optional) : maximum allowed depth of features.
    """

    # if just one entity, set max depth to 1 (transform stacking rule)
    if len(entityset.entity_dict) == 1 and (max_depth is None or max_depth > 1):
        warnings.warn("Only one entity in entityset, changing max_depth to "
                      "1 since deeper features cannot be created")
        max_depth = 1

    if selected_primitives is None:
        # use all primitives
    pass
```

**Comment by rwedge** [comment]
Should this return valid primitives for a specific target entity or after considering every entity as the target entity?  Do we want a tunable max_depth parameter?

**Comment by gsheni** [comment]
@rwedge good point. It should be for a specific target entity. I also added a max_depth parameter (defaults to 2 like DFS, but can be changed by the user).

**Comment by thehomebrewnerd** [comment]
Closed by #1462
